src/sentry/tsdb/snuba.py

```diff
@@ -442,32 +406,38 @@ def __get_data_snql(
                 conditions += model_query_settings.conditions
 
             project_ids = infer_project_ids_from_related_models(keys_map)
-            group_ids = keys_map.get("group_id")
-            env_ids = keys_map.get("environment")
-
-            translated_filter_keys = _translate_filter_keys(project_ids, group_ids, env_ids)
+            keys_map["project_id"] = project_ids
+            forward, reverse = get_snuba_translators(keys_map, is_grouprelease)
 
             # resolve filter_key values to the right values environment.id -> environment.name, etc.
-            mapped_filter_conditions = [
-                Condition(Column(k), Op.IN, list(v)) for k, v in translated_filter_keys.items()
-            ]
-            # map filter keys to conditional expressions
-            where_conds = (
-                [
+            mapped_filter_conditions = []
+            for col, f_keys in forward(deepcopy(keys_map)).items():
+                if f_keys:
+                    if len(f_keys) == 1 and None in f_keys:
```

Comment:
This looks like a change, was this happening previously but hidden in some extra abstraction layer?

Reply:
Yeah, this bit was hidden in the bowls of `bulk_raw_query` here: 
https://github.com/getsentry/sentry/blob/964343bac12fd57f82ff6b109ba91987a1d48db9/src/sentry/utils/snuba.py#L602-L607

The way I read it is for conditions like `["environment", "IN", ""]` went we translated the filters here:
https://github.com/getsentry/sentry/blob/964343bac12fd57f82ff6b109ba91987a1d48db9/src/sentry/utils/snuba.py#L1321

I think this bit was actually needed for cases where `environment.name == ""` like in this test case:
https://github.com/getsentry/sentry/blob/6c525637b84d15b884274f5b958ca1eeec004e50/tests/snuba/tsdb/test_tsdb_backend.py#L275-L289